### ISSUE TITLE
Different predicates on same path mess up routing

### ISSUE THREAD

**Issue by mariusgrigaitis** [zalando/skipper#1109]
**Describe the bug**
Defining same path but different predicates results in different routing

**To Reproduce**
```routes.eskip
glob:
    Path("/test/**") && Method("GET")
    -> inlineContent("/test/**")
    -> <shunt>;

root:
    Path("/test/") && Method("GET")
    -> inlineContent("/test/")
    -> <shunt>;

//post:
//    PathSubtree("/test") && Method("POST")
//    -> inlineContent("/test (POST)")
//   -> <shunt>;
```

```
GET /test -> 404
GET /test/ -> /test/
GET /test/foo/bar -> /test/**
```

After enabling post route:
```
GET /test -> /test/
GET /test/ -> /test/
GET /test/foo/bar -> /test/
```

**Comment by szuecs** [comment]
Yes, sorry `Path("/test/**")` is not what you expect. Basically there are two different "named" matches `:name` and `*anothername`. The special case `**` is a named match with the name `*`, not subtree glob as you expect and I guess is documented.

**Comment by mariusgrigaitis** [comment]
How about

```routes.eskip
named:
    Path("/test/:foo") && Method("GET")
    -> inlineContent("/test/:foo")
    -> <shunt>;

root:
    Path("/test/") && Method("GET")
    -> inlineContent("/test/")
    -> <shunt>;

//post:
//    PathSubtree("/test") && Method("POST")
//    -> inlineContent("/test (POST)")
//    -> <shunt>;
```

```
GET /test -> 404
GET /test/ -> /test/
GET /test/foo/bar -> /test/:foo
```

After enabling post route:
```
GET /test -> /test/
GET /test/ -> /test/
GET /test/foo/bar -> /test/:foo
```

**Comment by szuecs** [comment]
Ah now I got you, this looks like a bug, true.
It should be always 404 for the 1st case.

**Comment by aryszka** [comment]
fixed in #1200
